pkg/azure/client/storageaccount.go

```diff
@@ -52,21 +60,30 @@ func (c *StorageAccountClient) CreateStorageAccount(ctx context.Context, resourc
 	return err
 }
 
-// ListStorageAccountKey lists the first key of a storage account.
-func (c *StorageAccountClient) ListStorageAccountKey(ctx context.Context, resourceGroupName, storageAccountName string) (string, error) {
-	response, err := c.client.ListKeys(ctx, resourceGroupName, storageAccountName, &armstorage.AccountsClientListKeysOptions{
-		// doc: "Specifies type of the key to be listed. Possible value is kerb.. Specifying any value will set the value to kerb."
-		Expand: ptr.To("kerb"),
-	})
+// ListStorageAccountKeys lists all keys for the specified storage account.
+func (c *StorageAccountClient) ListStorageAccountKeys(ctx context.Context, resourceGroupName, storageAccountName string) ([]*armstorage.AccountKey, error) {
+	response, err := c.client.ListKeys(ctx, resourceGroupName, storageAccountName, &armstorage.AccountsClientListKeysOptions{})
 
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
-	if len(response.Keys) < 1 {
-		return "", fmt.Errorf("no key found in storage account %s", storageAccountName)
+	return response.Keys, nil
```

Comment:
https://github.com/gardener/gardener-extension-provider-azure/pull/1114#discussion_r2011774998

```suggestion
	return SortKeysByAge(response.Keys), nil
```